// Copyright 2021 Marco Nicola
// Copyright 2021 SpecializedGeneralist
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package api;

import "google/api/annotations.proto";

import "google/protobuf/descriptor.proto";

import "google/protobuf/empty.proto";

message ResponseErrors {
  repeated ResponseError value = 1;
}

message ResponseError {
  string message = 1;
}

message TranslateTextInput {
  string source_language = 1;

  string target_language = 2;

  string text = 3;
}

message TranslateTextResponse {
  TranslateTextData data = 1;

  ResponseErrors errors = 2;
}

message TranslateTextData {
  float took = 1;

  string translated_text = 2;
}

//TranslateTextParameters holds parameters to TranslateText
message TranslateTextRequest {
  TranslateTextInput translate_text_input = 1;
}

service Api {
  rpc TranslateText ( TranslateTextRequest ) returns ( TranslateTextResponse ) {
    option (google.api.http) = { post:"/translate_text" body:"translate_text_input"  };
  }
}
